Quicklook - QA data model

rkehoe opened the pull request:
This is an initial data model for the signal-to-noise QA for quicklook.

weaverba137 commented on doc/QUICKLOOK/snr_qa.rst:
```diff
@@ -0,0 +1,59 @@
+================ ============= ========== ============
+KEY              Example Value Type       Comment
+================ ============= ========== ============
+ARM              r             char       Spectrograph arm b,r,z
```
The spacing in this table looks a little off.

rkehoe replied:
Hello Ben,

In looking at the .rst file, and the web-output, it is not clear what 
spacing you are referring to?  Can you please clarify?

Bob

--------------------------------------------------------------------------
Robert Kehoe					      (214)768-1793 (off.)
Professor of Physics
Department of Physics				       (214)768-4095 (fax)
Southern Methodist University			     kehoe@physics.smu.edu
Dallas, TX 75275		          http://www.physics.smu.edu/kehoe

On Wed, 1 Jun 2016, Benjamin Alan Weaver wrote:

> 
> In doc/QUICKLOOK/snr_qa.rst:
> 
> > +
> > +
> > +Dictionary Contents
> > +===================
> > +
> > +KEYNAME = SNR
> > +
> > +Spectral signal-to-noise calculations per fiber and per CCD amp.
> > +
> > +Required Header Keywords
> > +~~~~~~~~~~~~~~~~~~~~~~~~
> > +
> > +================ ============= ========== ============
> > +KEY              Example Value Type       Comment
> > +================ ============= ========== ============
> > +ARM              r             char       Spectrograph arm b,r,z
> 
> The spacing in this table looks a little off.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[ASng8cPlCNvib2HMI7QPpSLXQdUCiD-lks5qHcLFgaJpZM4IrvK6.gif]
> 
> 
>

weaverba137 replied:
The columns need to be left-aligned with the table headers.  You want something like this:

```
==== ====
foo  bar
==== ====
a    b
==== ====
```

not

```
==== ====
foo  bar
==== ====
a   b
==== ====
```

I know that's a bit fussy, but that's how reStructuredText tables work.

rkehoe replied:
That is fine.  I just can't seem to see how to ensure that.  In the .rst 
file, the columns are aligned.  I do see that the html column headings are 
not all left justified, though.  How do you ensure it?

--------------------------------------------------------------------------
Robert Kehoe					      (214)768-1793 (off.)
Professor of Physics
Department of Physics				       (214)768-4095 (fax)
Southern Methodist University			     kehoe@physics.smu.edu
Dallas, TX 75275		          http://www.physics.smu.edu/kehoe

On Thu, 2 Jun 2016, Benjamin Alan Weaver wrote:

> 
> In doc/QUICKLOOK/snr_qa.rst:
> 
> > +
> > +
> > +Dictionary Contents
> > +===================
> > +
> > +KEYNAME = SNR
> > +
> > +Spectral signal-to-noise calculations per fiber and per CCD amp.
> > +
> > +Required Header Keywords
> > +~~~~~~~~~~~~~~~~~~~~~~~~
> > +
> > +================ ============= ========== ============
> > +KEY              Example Value Type       Comment
> > +================ ============= ========== ============
> > +ARM              r             char       Spectrograph arm b,r,z
> 
> The columns need to be left-aligned with the table headers. You want something
> like this:
> 
> ==== ====
> foo  bar
> ==== ====
> a    b
> ==== ====
> 
> not
> 
> ==== ====
> foo  bar
> ==== ====
> a   b
> ==== ====
> 
> I know that's a bit fussy, but that's how reStructuredText tables work.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[ASng8b5tZJoSGs7cqTINRzNwyuwlv0wwks5qHygrgaJpZM4IrvK6.gif]
> 
> 
>

weaverba137 replied:
I just took a look at the raw .rst file & it is properly aligned, so don't worry about this.

weaverba137 commented:
Typically, the filename of a data model file matches the filename it describes.  In this case it would be qa.rst.

weaverba137 commented:
I don't see a QUICKLOOK/index.rst file.  If you look at https://travis-ci.org/desihub/desidatamodel/jobs/134503347, you'll see its absence generates a warning.

sbailey commented:
Thanks for adding this documentation.  I think this is our first case of documenting a yaml file in the data model, and I'm not sure the mapping of how we did it for FITS files works very well for yaml.  i.e. I don't think I understand the structure of the yaml dictionary from reading this, and would find a commented example (with abbreviations for the long arrays) more useful, or useful as an additional example section to this doc.

weaverba137 commented on doc/QUICKLOOK/index.rst:
```diff
@@ -0,0 +1,8 @@
+.. toctree::
+   :maxdepth: 1
+
+   QUICKLOOK/index
```
This is a circular link.  Please take a look at the existing index.rst files in the data model to see how they work.

weaverba137 commented on doc/QUICKLOOK/qa-sky-CAMERA-EXPID.rst:
```diff
@@ -0,0 +1,73 @@
+
+Example YAML Output (10 spectra)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
This will look weird because it will not be in fixed width format.  See http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#literal-blocks for creating literal blocks (like `<pre>` in HTML).

rkehoe replied:
I believe the text is formatted fairly regularly. I have left the table 
as-is, and I have formatted the example beneath it.  Bob

--------------------------------------------------------------------------
Robert Kehoe					      (214)768-1793 (off.)
Professor of Physics
Department of Physics				       (214)768-4095 (fax)
Southern Methodist University			     kehoe@physics.smu.edu
Dallas, TX 75275		          http://www.physics.smu.edu/kehoe

On Thu, 2 Jun 2016, Benjamin Alan Weaver wrote:

> 
> In doc/QUICKLOOK/qa-sky-CAMERA-EXPID.rst:
> 
> > +================ ============= ========== ============
> > +KEY              Example Value Type       Comment
> > +================ ============= ========== ============
> > +ARM              r             char       Spectrograph arm b,r,z
> > +SPECTROGRAPH     0             int  	  Camera index 0..9
> > +EXPID            00000002      int  	  exposure ID
> > +QANAME		 SNR           string     name of QA algorithm
> > +TOTAL_SNR        44.484        float[500] summed S/N per fiber 
> > +MEDIAN_SNR       1.3233        float[500] median S/N per fiber
> > +TOT_AMP_SNR	 382.74	       float[4]   summed S/N averaged over each amplifier
> > +MEDIAN_AMP_SNR	 1.7905	       float[4]   median S/N averaged over each amplif
> ier
> > +================ ============= ========== ============
> > +
> > +Example YAML Output (10 spectra)
> > +~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> > +
> 
> This will look weird because it will not be in fixed width format. Seehttp://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#literal-bl
> ocks for creating literal blocks (like <pre> in HTML).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[ASng8RltEypZXMDj3eQveBCHCjJvG7pkks5qHzFIgaJpZM4IrvK6.gif]
> 
> 
>

rkehoe commented:
- QA .rst filename conforms to standard, i.e. corresponding .yaml data file name.
- added example output for QA as yaml example.  Keep prior table as it includes comments on example contents.
- added index.rst into QUICKLOOK, and added QUICKLOOK at top-level index.rst

weaverba137 commented:
This looks good to me.  @sbailey, does this satisfy your request for an example?

weaverba137 commented:
@sbailey, there's an outstanding question to  you on this PR, but otherwise it is ready to merge.  Please take a quick look when you get a chance.

sbailey commented:
Thanks for pinging me.  Yes, the example is helpful.  Merging now.
